rhizome/host/spec/cloud_hypervisor_spec.rb

```diff
@@ -13,23 +13,72 @@
       expect { fw.download }.to raise_error("Invalid SHA-256 digest")
     end
   end
+
+  describe "#cloud hypervisor" do
+    subject(:ch) { CloudHypervisor::VersionClass.new("35.1", "sha_ch", "sha_remote") }
+
+    it "can use an existing version of cloud hypervisor" do
+      expect(File).to receive(:exist?).with("/opt/cloud-hypervisor/v35.1/cloud-hypervisor").and_return(true)
+      expect(File).to receive(:exist?).with("/opt/cloud-hypervisor/v35.1/ch-remote").and_return(true)
+      expect(ch).not_to receive(:curl_file)
+      ch.download
+    end
+
+    it "can download cloud hypervisor" do
+      allow(Arch).to receive(:sym).and_return(:x64)
+      expect(File).to receive(:exist?).with("/opt/cloud-hypervisor/v35.1/ch-remote").and_return(false)
+      expect(File).to receive(:exist?).with("/opt/cloud-hypervisor/v35.1/cloud-hypervisor").and_return(false)
+      expect(FileUtils).to receive(:mkdir_p).with("/opt/cloud-hypervisor/v35.1").twice
+      expect(FileUtils).to receive(:chmod).with("a+x", "/opt/cloud-hypervisor/v35.1/ch-remote")
+      expect(FileUtils).to receive(:chmod).with("a+x", "/opt/cloud-hypervisor/v35.1/cloud-hypervisor")
+      f_remote = instance_double(File)
+      f_ch = instance_double(File)
+      expect(f_remote).to receive(:path).and_return("/opt/cloud-hypervisor/v35.1/ch-remote.tmp")
+      expect(f_ch).to receive(:path).and_return("/opt/cloud-hypervisor/v35.1/cloud-hypervisor.tmp")
+      expect(ch).to receive(:safe_write_to_file).with("/opt/cloud-hypervisor/v35.1/ch-remote").and_yield(f_remote)
+      expect(ch).to receive(:safe_write_to_file).with("/opt/cloud-hypervisor/v35.1/cloud-hypervisor").and_yield(f_ch)
+      expect(ch).to receive(:curl_file).with("https://github.com/cloud-hypervisor/cloud-hypervisor/releases/download/v35.1/ch-remote-static", "/opt/cloud-hypervisor/v35.1/ch-remote.tmp").and_return("sha_remote")
+      expect(ch).to receive(:curl_file).with("https://github.com/cloud-hypervisor/cloud-hypervisor/releases/download/v35.1/cloud-hypervisor-static", "/opt/cloud-hypervisor/v35.1/cloud-hypervisor.tmp").and_return("sha_ch")
+      ch.download
+    end
+
+    it "fails if sha is incorrect" do
+      allow(Arch).to receive(:sym).and_return(:x64)
+      expect(File).to receive(:exist?).with("/opt/cloud-hypervisor/v35.1/ch-remote").and_return(false)
+      expect(File).to receive(:exist?).with("/opt/cloud-hypervisor/v35.1/cloud-hypervisor").and_return(false)
+      expect(FileUtils).to receive(:mkdir_p).with("/opt/cloud-hypervisor/v35.1").twice
+      expect(FileUtils).to receive(:chmod).with("a+x", "/opt/cloud-hypervisor/v35.1/ch-remote")
+      expect(FileUtils).not_to receive(:chmod).with("a+x", "/opt/cloud-hypervisor/v35.1/cloud-hypervisor")
+      f_remote = instance_double(File)
+      f_ch = instance_double(File)
```

Comment:
You could use `instance_double(File, path: 'blah', some_other_attr: 'whatever')`